### PR TITLE
feat: add create match note from URL command

### LINKED ISSUE #21: Implement "Create match note from URL" command
## Summary
Add the first real user-facing command for the plugin: prompt for a match URL and start the match-note creation flow.

## Why
The current MVP backlog has schema, parsing, templating, and tests, but it still needs the actual command that a user runs from Obsidian. This is the entry point that turns the plugin into a usable workflow.

## Scope
- Register a command with a stable command id.
- Prompt the user for a URL.
- Pass the input into the URL parsing / validation layer.
- Pass the parsed result into the note-creation flow.
- Show useful notices for success or failure.

## Implementation expectations
- Keep the command entry point thin.
- Delegate parsing and note generation to separate modules.
- Do not embed provider-specific fetching in this command.
- Keep user-facing copy short and clear.

## Proposed command
- Command id: `create-match-note-from-url`
- Command name: `Create match note from URL`

## Definition of done
- A user can run the command from the command palette.
- The plugin prompts for a URL and runs the match-note creation flow.
- Errors are surfaced cleanly without crashing the plugin.

### PR DESCRIPTION
**Pull request by FerdiHS**
This pull request introduces a new "Create match note from URL" workflow, including its command registration, modal UI, and comprehensive tests. The workflow allows users to paste a football match URL and automatically generate a match note. The implementation is modular, testable, and integrates with the plugin's command system. The most important changes are grouped below.

**Feature: Create Match Note from URL Workflow**

* Added `create-match-note-from-url.ts` implementing the workflow, including error handling, dependency injection, and command registration logic.
* Registered the new command in the plugin lifecycle via `registerCommands` in `main.ts` and `commands/index.ts`. [[1]](diffhunk://#diff-a6a1db61476382a80351bf0b40126b3cfca87878293a3fca70dfe04367f4a11fR1-R6) [[2]](diffhunk://#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dR2) [[3]](diffhunk://#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dR11)

**User Interface: Match URL Modal**

* Implemented `MatchUrlModal` in `match-url-modal.ts`, providing a modal dialog for users to input match URLs, with proper submission, validation, and button state handling.
* Added `canCloseMatchUrlModal` utility and corresponding test to control modal dismissal during submission. [[1]](diffhunk://#diff-a206f59fe90590fc0bf392a182de3f958ecae4635ffce7ac8578c3bd4c7405cfR1-R3) [[2]](diffhunk://#diff-a63d1950620ee0b54468bb60fe50e3d0a2bfab244de443b56c09996fc19a901eR1-R9)

**Testing and Build Improvements**

* Added comprehensive tests for the new workflow, command registration, and modal state logic. [[1]](diffhunk://#diff-efdb49f30680992431728e73450601a973b9bf97d7d65a2b01076596dac2675eR1-R220) [[2]](diffhunk://#diff-a63d1950620ee0b54468bb60fe50e3d0a2bfab244de443b56c09996fc19a901eR1-R9)
* Updated the `test` script in `package.json` to include new test files for full coverage.

**Related Issues:**

* Resolve #21

### CODE REVIEW

**Comment by FerdiHS** [comment]
@codex Please do a final review of this PR.

Check for coding bugs, logic bugs, dead code, bad patterns, edge cases, typo issues, and missing test coverage. If everything looks good, please confirm whether this PR is ready and safe to merge.

File: src/commands/create-match-note-from-url.test.ts [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+import { type App, type TFile } from 'obsidian';
+
+import type FootballNotesPlugin from '../main';
+import { parseMatchUrl } from '../services/match-url-parser';
+import {
+	CREATE_MATCH_NOTE_FROM_URL_COMMAND_ID,
+	CREATE_MATCH_NOTE_FROM_URL_COMMAND_NAME,
+	createMatchNoteFromUrl,
+	registerCreateMatchNoteFromUrlCommand,
+} from './create-match-note-from-url';
```
[the remark this answers is not in the document]
Comment (FerdiHS) [reply src/commands/create-match-note-from-url.test.ts:13]: Tracked in #29 as a follow-up refactor. This PR keeps the current Obsidian adapter in place so the user-facing behavior stays unchanged, and the current test coverage remains valid for this scope.

**Comment by FerdiHS** [comment]
@codex Please do a final review of this PR again.

I’ve decided to keep the Obsidian-free split as a follow-up in #29, so this PR stays focused on the user-facing command flow. Please re-check for coding bugs, logic bugs, dead code, bad patterns, edge cases, typo issues, and missing test coverage, and confirm whether it is ready and safe to merge.

File: package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
 	"scripts": {
 		"dev": "node esbuild.config.mjs",
 		"build": "tsc -noEmit -skipLibCheck && node esbuild.config.mjs production",
-		"test": "node --import jiti/register --test src/services/match-url-parser.test.ts",
+		"test": "node --import jiti/register --test src/services/match-url-parser.test.ts src/commands/create-match-note-from-url.test.ts src/ui/match-url-modal-state.test.ts",
```
[the remark this answers is not in the document]
Comment (FerdiHS) [reply package.json:10]: I verified the current test path and the reported failure does not reproduce in this PR.

The command module still has a real `Notice` import, but the Node test suite is already using `jiti/register`, and `npm test` passes with the current `create-match-note-from-url.test.ts` coverage. I’m keeping the Obsidian-free split as follow-up work in #29 so this PR stays focused on the user-facing command flow.

**Comment by FerdiHS** [comment]
@codex Please do one more review of this PR and report all issues you find, not just major ones.

Please check for coding bugs, logic bugs, dead code, bad patterns, edge cases, typo issues, and missing test coverage. Please also keep #29 in mind as a planned follow-up refactor, so only flag issues that should be fixed in this PR. If everything looks good, please confirm whether this PR is ready and safe to merge.

**Comment by FerdiHS** [comment]
@codex please review the full PR again after the latest fix. Report every issue you find, not just major ones, including grammar, clarity, missing details, misleading wording, and edge cases. If there are no issues, please say so explicitly.

File: src/ui/match-url-modal.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+		contentEl.empty();
+		this.setTitle('Create match note from URL');
+
+		contentEl.createEl('p', {
+			text: 'Paste a football match URL to create a new match note.',
+		});
+
+		const inputContainer = contentEl.createDiv({ cls: 'football-notes-match-url-input' });
+		this.matchUrlInput = new TextComponent(inputContainer);
+		this.matchUrlInput.setPlaceholder('https://example.com/match/123');
```
[the remark this answers is not in the document]
Comment (FerdiHS) [reply src/ui/match-url-modal.ts:40]: Fixed. Added a visible `Match URL` label and associated it with the input so the modal has a stable accessible name instead of relying on placeholder text alone.

File: src/commands/create-match-note-from-url.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (FerdiHS) [reply src/commands/create-match-note-from-url.test.ts:13]: Fixed. The command module no longer imports `Notice` at module evaluation time; it now lazy-loads the Obsidian runtime only when the default notice path is actually used. This keeps the Node test import path clean while preserving the current behavior.

**Comment by FerdiHS** [comment]
@codex Please do one more review of this PR and report all issues you find, not just major ones.

Please check for coding bugs, logic bugs, dead code, bad patterns, edge cases, typo issues, and missing test coverage. Please also keep #29 in mind as a planned follow-up refactor, so only flag issues that should be fixed in this PR. If everything looks good, please confirm whether this PR is ready and safe to merge.
